chart_utils.py

```diff
@@ -0,0 +1,244 @@
+                    "SithSwap_debt_token_supply",
+                    "JediSwap_debt_token_supply",
+                    "debt_token_supply",
+                ]:
+                    combined_df[col] += combined_df[f"{col}_y"]
+
+                # Drop the '_y' columns after summing the relevant values
+                combined_df.drop(
+                    [col for col in combined_df.columns if col.endswith("_y")],
+                    axis=1,
+                    inplace=True,
+                )
+
+        # Create a new pair name for the stablecoin bundle
+        new_pair = f"{collateral}-{src.settings.STABLECOIN_BUNDLE_NAME}"
+        # Add the combined DataFrame for this collateral to the data dictionary
+        data[new_pair] = combined_df
+
+    # Return the updated data dictionary
+    return data
+
+
+@streamlit.cache_data(ttl=300)
+def get_load_data(
+    protocol_name: str,
+) -> Tuple[Dict[str, pd.DataFrame], Dict[str, pd.DataFrame]]:
+    """
+    Load loan data and main chart data for the specified protocol.
+    :param protocol_name: Protocol name.
+    :return: DataFrames containing loan data and main chart data.
+    """
+    return src.helpers.load_data(protocol=protocol_name)
+
+
+def get_protocol_data_mappings(
+    current_pair: str, stable_coin_pair: str, protocols: list[str]
+) -> Tuple[Dict[str, dict], Dict[str, dict]]:
+    """
+    Get protocol data mappings for main chart data and loans data.
+
+    :param current_pair: The current pair for which data is to be fetched.
+    :param stable_coin_pair: The stable coin pair to check against.
+    :param protocols: List of protocols for which data is to be fetched.
+    :return: tuple of dictionaries containing:
+        - protocol_main_chart_data: Mapping of protocol names to their main chart data.
+        - protocol_loans_data: Mapping of protocol names to their loans data.
+    """
+
+    protocol_main_chart_data: Dict[str, dict] = {}
+    protocol_loans_data: Dict[str, dict] = {}
+
+    for protocol_name in protocols:
+        main_chart_data, loans_data = get_load_data(protocol_name)
+        protocol_loans_data[protocol_name] = loans_data
+
+        if current_pair == stable_coin_pair:
+            protocol_main_chart_data[protocol_name] = create_stablecoin_bundle(
+                main_chart_data
+            )[current_pair]
+        else:
+            protocol_main_chart_data[protocol_name] = main_chart_data[current_pair]
+
+    return protocol_main_chart_data, protocol_loans_data
+
+
+def load_stats_data() -> (
+    Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame, pd.DataFrame]
+):
+    """
+    Load general stats, supply stats, collateral stats, and debt stats data.
+    :return: tuple of DataFrames containing general stats, supply stats, collateral stats, debt stats, and utilization stats.
+    """
+    BASE_GS_PATH = f"gs://{src.helpers.GS_BUCKET_NAME}/data"
+    read_parquet_with_protocol_index = partial(pd.read_parquet, engine="fastparquet")
```

Comment:
Why not specify `engine="fastparquet"` in `read_and_set_index` directly?

Reply:
I'm trying to following DRY(Don't repeat yourself) Instead of having this line a lot of times, I defined it only once.